File: CS/CustomFilterPopupExample/CustomGridFilterColumn/CustomGridFilterColumn.cs
using DevExpress.Data.Filtering.Helpers;
using DevExpress.Xpf.Grid;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CustomFilterPopupExample
{
    class CustomGridFilterColumn : GridFilterColumn
    {
        public CustomGridFilterColumn(ColumnBase column, bool useDomainDataSourceRestrictions, bool useWcfSource)
            : base(column, useDomainDataSourceRestrictions, useWcfSource) { }

        public override bool IsValidClause(DevExpress.Data.Filtering.Helpers.ClauseType clause)
        {
            if (this.ColumnType == typeof(int))
            {
                if (clause == ClauseType.Equals || clause == ClauseType.Greater || clause == ClauseType.GreaterOrEqual || clause == ClauseType.Less || clause == ClauseType.LessOrEqual || clause == ClauseType.DoesNotEqual)
                {
                    return true;
                }
                else return false;
            }
            else if (this.ColumnType == typeof(string))
            {
                if (clause == ClauseType.Equals || clause == ClauseType.BeginsWith || clause == ClauseType.EndsWith || clause == ClauseType.Contains || clause == ClauseType.DoesNotEqual)
                {
                    return true;
                }
                else return false;
            }
            else return true;
        }
    }
}

File: VB/CustomFilterPopupExample/CustomGridFilterColumn/CustomGridFilterColumn.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Data.Filtering.Helpers
Imports DevExpress.Xpf.Grid
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace CustomFilterPopupExample
	Friend Class CustomGridFilterColumn
		Inherits GridFilterColumn
		Public Sub New(ByVal column As ColumnBase, ByVal useDomainDataSourceRestrictions As Boolean, ByVal useWcfSource As Boolean)
			MyBase.New(column, useDomainDataSourceRestrictions, useWcfSource)
		End Sub

		Public Overrides Function IsValidClause(ByVal clause As DevExpress.Data.Filtering.Helpers.ClauseType) As Boolean
			If Me.ColumnType Is GetType(Integer) Then
                If clause = ClauseType.Equals OrElse clause = ClauseType.Greater OrElse clause = ClauseType.GreaterOrEqual OrElse clause = ClauseType.Less OrElse clause = ClauseType.LessOrEqual OrElse clause = ClauseType.DoesNotEqual Then
                    Return True
                Else
                    Return False
                End If
			ElseIf Me.ColumnType Is GetType(String) Then
                If clause = ClauseType.Equals OrElse clause = ClauseType.BeginsWith OrElse clause = ClauseType.EndsWith OrElse clause = ClauseType.Contains OrElse clause = ClauseType.DoesNotEqual Then
                    Return True
                Else
                    Return False
                End If
			Else
				Return True
			End If
		End Function
	End Class
End Namespace
